common/junit-platform-native/src/main/java/org/graalvm/junit/platform/config/NativeImageConfigurationImpl.java

```diff
@@ -0,0 +1,85 @@
+ * copyright rights in the Software, and any and all patent rights owned or
+ * freely licensable by each licensor hereunder covering either (i) the
+ * unmodified Software as contributed to or provided by such licensor, or (ii)
+ * the Larger Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ *
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ *
+ * The above copyright notice and either this complete permission notice or at a
+ * minimum a reference to the UPL must be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.graalvm.junit.platform.config;
+
+import org.graalvm.junit.platform.JUnitPlatformFeature;
+import org.graalvm.junit.platform.config.core.NativeImageConfiguration;
+import org.graalvm.nativeimage.hosted.RuntimeClassInitialization;
+import org.graalvm.nativeimage.hosted.RuntimeReflection;
+
+import java.lang.reflect.Executable;
+import java.lang.reflect.Field;
+
+public class NativeImageConfigurationImpl extends NativeImageConfiguration {
+
+    @Override
+    public void registerForReflection(Class<?>... classes) {
+        RuntimeReflection.register(classes);
+    }
+
+    @Override
+    public void registerForReflection(Executable... methods) {
+        RuntimeReflection.register(methods);
+    }
+
+    @Override
+    public void registerForReflection(Field... fields) {
+        RuntimeReflection.register(fields);
+    }
+
+    @Override
+    public void initializeAtBuildTime(String... classNames) {
+        for (String className : classNames) {
+            Class<?> clazz;
+            try {
+                clazz = Class.forName(className);
+                initializeAtBuildTime(clazz);
+            } catch (ClassNotFoundException e) {
+                JUnitPlatformFeature.debug(" [Native Image Configuration] Failed to register class for build time initialization: %s Reason: %s", className, e);
```

Comment:
It was intended - it seemed a bit nicer to separate this particular log with two spaces:
```
[Debug] Some other log
[Debug]  [Native Image Configuration] A message
```
However, I can remove it

Reply:
Yes, please remove it.

Let's not focus on special formatting here.

Ideally we should be using a logging framework that allows the user to specify formatting of log messages.

So let's just pretend we are already using a logging framework, since that will allow us to migrate to a logging framework more easily later.

Reply:
I will remove the space.

I've also given a logging framework a thought while implementing this, but unfortunately, due to the context in which we are running, we can't safely do that: any code that runs in the feature implicitly initializes all used classes at image build time. If we were to use a logging framework, we would force the initialization of all used logging framework classes - this would affect user applications

Reply:
Hmmm..... what is the recommend approach for logging from a `Feature`?

/cc @vjovanov

Reply:
I've seen logging be one of the most common pain points with build time initialization - let's say you're using SLF4J and create a logger in a static field in a class that's initialized at build time: depending on what the user plugs in as the underlying logging framework, you will initialize different classes at build time. This also means async logging/writing output to a file become unusable as the thread for logging would get started at image build time (in the builder JVM) and the file would be created on the machine building the image.

One thing I've just checked is that we initialize the `java.util` JDK package at build time - this would include the JDK logging framework under `java.util.logging`. In theory, this should mean we could safely use it for now, hopefully without affecting third party applications (this depends on how much static state from the image build remains in these classes). Another alternative would be a mini-homegrown logging framework, but I would avoid that for now